calico_node/allocate_ipip_addr/allocate_ipip_addr.go

```diff
@@ -0,0 +1,112 @@
+package allocateIPIPAddr
+
+import (
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/projectcalico/libcalico-go/lib/api"
+	"github.com/projectcalico/libcalico-go/lib/client"
+	"debug/pe"
+)
+
+func allocateIPIPAddr(client *client.Client, string nodename) {
+	meta := api.IPPoolMetadata{}
+	ipPoolList, err := client.IPPools().List(meta)
+	if err != nil {
+		log.Warnf("error retrieving pools: %s", err)
+	}
+    ipv4Pools := getIPv4Pools(ipPoolList.Items)
+    enabledPools := getIPIPEnabledPools(ipv4Pools)
+    if len(enabledPools) > 0 {
```

Comment:
I don't think we need both IPv4 and IPIP pools - it should be sufficient to just use the list of IPIP pools - if the tunl address is not in an IPIP pool then remove it and assign an address from an IPIP pool.